### PR TITLE
PKX-485 Added user bulk registration

### PR DESCRIPTION
**Pull request by mubtadaali**
### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: openedx/features/pakx/lms/pakx_admin_app/views.py [hunk 1]
```diff
@@ -41,19 +43,20 @@
...
     get_org_users_qs,
     get_roles_q_filters,
+    get_user_data_from_bulk_registration_file,
     get_user_org,
     get_user_org_filter,
     get_user_same_org_filter,
     is_courses_enroll_able,
-    is_courses_user_have_same_org,
-    send_registration_email
+    is_courses_user_have_same_org
```
Comment (emzubair) [inline openedx/features/pakx/lms/pakx_admin_app/views.py:59]: we should refactor this method name in near future, `do_courses_and_user_have_same_org`

File: openedx/features/pakx/lms/pakx_admin_app/views.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
 """
 Views for Admin Panel API
 """
+import csv
+import io
```
Comment (emzubair) [inline openedx/features/pakx/lms/pakx_admin_app/views.py:5]: Can we import `StringIO` and `DictReader` respectively from the above imports?

File: openedx/features/pakx/lms/pakx_admin_app/views.py [hunk 3]
```diff
@@ -159,15 +162,28 @@ def get_serializer_class(self):
...
+    def bulk_registration(self, request, *args, **kwargs):
+        if not request.FILES.get('file'):
+            return Response('File is required!', status=status.HTTP_400_BAD_REQUEST)
+
+        file_data = io.StringIO(request.FILES['file'].read().decode('utf-8'))
+        file_reader = csv.DictReader(file_data)
+
+        required_col_names = {'name', 'username', 'email', 'organization_id', 'role', 'employee_id', 'language'}
+        if not set(file_reader.fieldnames) == required_col_names:
+            return Response('Invalid file format! Column names are incorrect!', status=status.HTTP_400_BAD_REQUEST)
```
Comment (emzubair) [inline openedx/features/pakx/lms/pakx_admin_app/views.py:181]: Would suggest returning info about the correct format, i.e `"Invalid Column names, valid format:{}".format(required_col_names)`.
Thoughts?

File: openedx/features/pakx/lms/pakx_admin_app/views.py [hunk 4]
```diff
@@ -159,15 +162,28 @@ def get_serializer_class(self):
...
+
+        users = get_user_data_from_bulk_registration_file(file_reader, self.request.user.profile.organization_id)
```
Comment (emzubair) [inline openedx/features/pakx/lms/pakx_admin_app/views.py:183]: Should we introduce a util method to get current user org, we've 3 places where we are using the same line of code?
`get_request_user_org(self.request)`

File: openedx/features/pakx/lms/pakx_admin_app/views.py (hunk 1 above)
Comment (mubtadaali) [reply openedx/features/pakx/lms/pakx_admin_app/views.py:59]: how about `is_courses_and_user_have_same_org`?

File: openedx/features/pakx/lms/pakx_admin_app/views.py (hunk 3 above)
Comment (mubtadaali) [reply openedx/features/pakx/lms/pakx_admin_app/views.py:181]: A sample google sheet will be shared will them, also add all column names in that notification will make the message very length, ?

Comment (emzubair) [reply openedx/features/pakx/lms/pakx_admin_app/views.py:182]: @mubtadaali I agree, but we've often such mistakes from the Authoring end, so I would suggest sending the valid format.

**Pull request merged**